### PR TITLE
fix: chunk entities to stay under 4MB

### PR DESCRIPTION
**Pull request by jajeffries**
This pull request adds comprehensive tests and new chunking logic to the policy runner, improving how entities are ingested and tested. The main changes include implementing chunked ingestion for large entity lists, updating related tests to cover chunking and error scenarios, and adding utility methods for chunking and size estimation.

**Policy runner ingestion logic improvements:**

* Added chunking logic to the `run` method in `runner.py` to split entities into 3.5MB chunks for ingestion, with detailed logging and error handling for each chunk. (`[worker/worker/policy/runner.pyL120-R133](diffhunk://#diff-77ccd0c70b117c80a734aad5873b3fce2bfca6d5d1486c8bcf12f6e19cdd57e2L120-R133)`)
* Implemented `_create_message_chunks` and `_estimate_message_size` methods in `runner.py` to support chunked ingestion and accurate message size calculation. (`[worker/worker/policy/runner.pyR182-R212](diffhunk://#diff-77ccd0c70b117c80a734aad5873b3fce2bfca6d5d1486c8bcf12f6e19cdd57e2R182-R212)`)

**Test suite enhancements:**

* Updated existing tests in `test_runner.py` to use mock entities and verify chunked ingestion behavior, including error handling and logging for individual chunks. (`[[1]](diffhunk://#diff-f6ad41627bb040e8e4b7e13d214a73c2bd73b5da5f9882ec7bfe9b1c83c63a51R166-R185)`, `[[2]](diffhunk://#diff-f6ad41627bb040e8e4b7e13d214a73c2bd73b5da5f9882ec7bfe9b1c83c63a51R198-R206)`, `[[3]](diffhunk://#diff-f6ad41627bb040e8e4b7e13d214a73c2bd73b5da5f9882ec7bfe9b1c83c63a51L194-R218)`, `[[4]](diffhunk://#diff-f6ad41627bb040e8e4b7e13d214a73c2bd73b5da5f9882ec7bfe9b1c83c63a51R283-R291)`, `[[5]](diffhunk://#diff-f6ad41627bb040e8e4b7e13d214a73c2bd73b5da5f9882ec7bfe9b1c83c63a51L272-R303)`, `[[6]](diffhunk://#diff-f6ad41627bb040e8e4b7e13d214a73c2bd73b5da5f9882ec7bfe9b1c83c63a51L319-R350)`)
* Added new tests for `_create_message_chunks` and `_estimate_message_size` covering empty lists, single/multiple chunks, edge cases, and size estimation. (`[worker/tests/policy/test_runner.pyR368-R547](diffhunk://#diff-f6ad41627bb040e8e4b7e13d214a73c2bd73b5da5f9882ec7bfe9b1c83c63a51R368-R547)`)
* Added a test for the `run` method with multiple chunks and chunk-specific error handling, verifying correct logging and ingest calls. (`[worker/tests/policy/test_runner.pyR368-R547](diffhunk://#diff-f6ad41627bb040e8e4b7e13d214a73c2bd73b5da5f9882ec7bfe9b1c83c63a51R368-R547)`)

**Imports and setup:**

* Added import for `ingester_pb2` in both `runner.py` and `test_runner.py` to support entity creation and size estimation. (`[[1]](diffhunk://#diff-f6ad41627bb040e8e4b7e13d214a73c2bd73b5da5f9882ec7bfe9b1c83c63a51R9)`, `[[2]](diffhunk://#diff-77ccd0c70b117c80a734aad5873b3fce2bfca6d5d1486c8bcf12f6e19cdd57e2R8-R14)`)

### CODE REVIEW

File: worker/worker/policy/runner.py [hunk 1]
```diff
@@ -117,7 +121,18 @@ def run(
...
+                chunk_size_mb = self._estimate_message_size(entity_chunk) / (1024 * 1024)
+                logger.info(
+                    f"Ingesting chunk {chunk_num} with {len(entity_chunk)} entities (~{chunk_size_mb:.2f} MB)"
+                )
+                response = client.ingest(entities=entity_chunk)
+                if response.errors:
+                    raise RuntimeError(f"Chunk {chunk_num} ingestion failed: {response.errors}")
+                logger.info(f"Chunk {chunk_num} ingested successfully")
+
+            logger.info("Successfully ingested all entities into NetBox")
```
Comment (leoparente) [inline worker/worker/policy/runner.py]: remove 'into Netbox' as it is not directly ingesting it there. Maybe, add the number of ingested entities

Comment (jajeffries) [reply worker/worker/policy/runner.py]: Done!

File: worker/worker/policy/runner.py [hunk 2]
```diff
@@ -117,10 +121,18 @@ def run(
...
         try:
             entities = backend.run(self.name, policy)
-            response = client.ingest(entities)
-            if response.errors:
-                raise Exception(f"Ingestion failed with errors: {response.errors}")
-            logger.info(f"Policy {self.name}: Successful ingestion")
+
+            for chunk_num, entity_chunk in enumerate(self._create_message_chunks(entities), 1):
+                chunk_size_mb = self._estimate_message_size(entity_chunk) / (1024 * 1024)
+                logger.info(
```
Comment (leoparente) [inline worker/worker/policy/runner.py]: Maybe change this to be debug log

File: worker/worker/policy/runner.py [hunk 3]
```diff
@@ -117,10 +121,18 @@ def run(
...
+                    f"Ingesting chunk {chunk_num} with {len(entity_chunk)} entities (~{chunk_size_mb:.2f} MB)"
+                )
+                response = client.ingest(entities=entity_chunk)
+                if response.errors:
+                    raise RuntimeError(f"Chunk {chunk_num} ingestion failed: {response.errors}")
+                logger.info(f"Chunk {chunk_num} ingested successfully")
```
Comment (leoparente) [inline worker/worker/policy/runner.py]: Maybe change this to be debug log

**Review by leoparente — APPROVED** [review]
